src/services/ui/src/features/package-actions/ActionPage.tsx

```diff
@@ -1,22 +1,39 @@
 import { BreadCrumbs, FAQFooter, SimplePageContainer } from "@/components";
-import { useParams } from "@/components/Routing";
 import { getSetupFor } from "@/features/package-actions/lib";
 import { ActionForm } from "@/features/package-actions/ActionForm";
 import { detailsAndActionsCrumbs } from "@/utils";
+import { Navigate, useParams } from "react-router-dom";
+import { Action, Authority, AuthorityUnion } from "shared-types";
 
 export const ActionPage = () => {
-  const { id, type, authority } = useParams("/action/:authority/:id/:type");
+  const {
+    id,
+    type: actionType,
+    authority,
+  } = useParams<{ id: string; type: Action; authority: Authority }>();
+
+  // TODO: use zod
+  if (!id || !actionType || !authority) {
+    return <Navigate to="/dashboard" />;
+  }
+
+  const setup = getSetupFor(actionType, authority as AuthorityUnion);
```

Comment:
Casting because `authority` is typed as `Authority` (lowercase values) but actually has normal case values (since it comes from the params). 
Hope this doesn't confuse anyone. Perhaps we can standardize this in the future.